paychmgr/paych.go

```diff
@@ -417,3 +417,27 @@ func (ca *channelAccessor) settle(ctx context.Context, ch address.Address) (cid.
 
 	return smgs.Cid(), err
 }
+
+func (ca *channelAccessor) collect(ctx context.Context, ch address.Address) (cid.Cid, error) {
+	ca.lk.Lock()
+	defer ca.lk.Unlock()
+
+	ci, err := ca.store.ByAddress(ch)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	msg := &types.Message{
+		To:     ch,
+		From:   ci.Control,
```

Comment:
I just moved the existing code here, but I'm not sure if the From address should always be ChannelInfo.Control?
Presumably only if the node is the creator of the channel (ie if ChannelInfo.Outbound == true)

The TestPaymentChannels integration test passes even though the receiver is the one who collects. Which makes me wonder if `MpoolPushMessage()` actually uses the sending node's address as the From address in the message.

Reply:
@arajasek points out that the Control address is always the "local" address so the code is correct.
Note that specs-actors accepts settle and collect messages from either party to the payment channel (the creator or the recipient)